packages/embed/src/js/basic-kyc.js

```diff
@@ -855,10 +855,14 @@ import { version as sdkVersion } from '../../package.json';
     const message = userTriggered
       ? 'SmileIdentity::Close'
       : 'SmileIdentity::Close::System';
-    referenceWindow.postMessage(message, '*');
+    [referenceWindow.parent, referenceWindow].forEach((win) => {
```

Comment:
There are situations where the parent listens for the message and other times that the grand parent does the listening. This is to ensure that both the parent and grand parent receive the message.

Note: to find a better approach